### ISSUE TITLE
Discussão das regras de negócio do He4rt Squads

### ISSUE THREAD

**Issue by lanjoni** [he4rt/he4rt-bot-api#60]
# He4rt Squad

O projeto busca desenvolver uma plataforma para criação de squads para membros da He4rt Developers, buscando aplicar conceitos de hackathons e agrupação de membros no geral.

## Sobre os Squads
Informações gerais sobre como o sistema de squads deverá funcionar.

### Tipos de Squads

O projeto não é visado apenas para times da Hackathon, mas no geral de formar grupos de amigos que querem contribuir dentro da comunidade. Logo, os tipos de squads são:

- Geral - Contribuidores de projetos internos e externos providenciados pela comunidade;
- Evangelizadores - Pessoas que levam a palavra da He4rt dentro de ambientes em comum;
- Competidores - Pessoas que se preparam pra representar a He4rt dentro de Hackathons;

### Cargos Esperados por Squad 

Os cargos esperados não contam com o Líder/Sub Líder

- 1 (um) Pessoa de Front-end 
- 1 (um) Pessoa de Back-end
- 1 (um) Pessoa de DevOPS (opcional) 
- 1 (um) Pessoa de UI/UX (opcional) 
- 1 (um) Pessoa de QA (opcional) 
- 1 (um) Pessoa de Negócios (opcional) 

### Sobre o Líder

- Toda squad tem um líder;
- Toda squad pode ter um sub-líder
- Ao criar uma squad o usuário é atribuído como líder
- O líder pode repassar seu título
- O líder pode nomear um sub-líder
- O líder representa a squad em meios de comunicação
- O líder é responsável pela sua squad e entregas
- Apenas o líder pode enviar convites para a squad

### Fluxo de aplicação:

- Um usuário poderá criar um grupo com o nome que quiser;
- Esse grupo terá algumas regras para que seja "ATIVO/VERIFICADO"
    - Quantidade minima de membros (5);
    - Level minimo de membros da squad;
    - Líder com level minimo dentro do servidor;
- Após criar, o líder poderá convidar outros membros seguindo as regras:
    - Não pertencer à nenhuma outra squad;
    - TODO: a decidir
- Feita todas as etapas, terá um grupo oficial para representar a He4rt.

**Comment by gitlherme** [comment]
Alguns pontos que eu acho que possa fazer sentido.

### Tipos de squads
Squads é uma entidade geral, não precisa possuir um tipo específico. Dentro de cada squad voce pode ter subgrupos, como o geral, evangelizador e competidor.

Isso pode ajudar e trazer mais interação entre pessoas com objetivos iniciais diferentes, ou que querem fazer algo mas ainda estão em dúvida. É interessante que a pessoa possa transicionar entre diferentes partes do Squad pra ver o que mais faz sentido pra ela e onde ela quer ajudar.

### Cargos Esperados por Squad
Faria sentido isso ser direcionado apenas para a parte de competição dos squads? Uma pessoa que é apenas curiosa, poderia participar de um squad sem ter necessariamente um cargo?

### Regras de negócio
Fluxo de aplicação: Um usuário poderá criar um grupo com o nome que quiser; - aqui seria interessante manter um padrão? Digo, se a criação de um squad não passar por uma moderação, ser algo automático, pode-se criar nomes mal intencionados, zoeirinhos ou qualquer coisa do tipo que possa diminuir a inclusão dentro das squads. Por exemplo, ter um nome onde uma pessoa trans ou não binária, não se sentisse a vontade de participar.

**Comment by fellipesena** [comment]
Acho que faria sentido um membro participar de mais de um squad, deixando um relacionamento n x n.

Digo isso para membros que querem participar de dois eventos que acontecerão em dias próximos, para não ter que ficar trocando de squad em squad.

Acha que é uma ideia válida?

**Comment by Luisnadachi** [comment]
Concordo com a questão dos **Tipos de Squads** que o @gitlherme trouxe, a gente ter uma squad maior, talvez a ver com o estado da pessoa (1:N) enquanto ela pode participar de varios times dentro da squad (N:N). Sobre os cargos, acho que podemos manter os cargos mas não limitar a quantidade de cargos nas equipes. Já a regra de negócio, concordo 100% com o ponto levantado, não imagino além de padrão ou talvez validação com regex como podemos evitar nomes problemáticos, mas isso dai é uma coisa a se pensar mesmo
